contracts/HubPool.sol

```diff
@@ -370,39 +392,61 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
     }
 
     /**
-     * @notice Whitelist an origin chain ID + token <-> destination token route. Callable only by owner.
-     * @param originChainId Chain where deposit occurs.
-     * @param destinationChainId Chain where depositor wants to receive funds.
-     * @param originToken Deposited token.
-     * @param destinationToken Token that depositor wants to receive on destination chain. Unused if `enableRoute` is
-     * False.
-     * @param enableRoute Set to true to enable route on L2 and whitelist new destination token, or False to disable
-     * route on L2 and delete destination token mapping on this contract.
+     * @notice Stores mapping from origin chain ID + token + destination chain ID to token. This mapping corresponds
+     * to token transfer routes and can be used for both HubPool => SpokePool rebalances and SpokePool ==> SpokePool
+     * deposits. Callable only by owner.
+     * @param originChainId Chain where token transfer originates.
+     * @param destinationChainId Chain where token transfer ends up.
+     * @param originToken Token sent from origin chain.
+     * @param destinationToken Token received at destination chain.
+     * @param depositsEnabled Set to true to whitelist this route for deposits, set to false if caller just wants to
+     * map the origin token + origin ID + destination ID to the destination token address and wants to explicitly
+     * disable this route for deposits.
      */
+
     function whitelistRoute(
         uint256 originChainId,
         uint256 destinationChainId,
         address originToken,
         address destinationToken,
-        bool enableRoute
-    ) public override onlyOwner nonReentrant {
-        if (enableRoute)
-            whitelistedRoutes[_whitelistedRouteKey(originChainId, originToken, destinationChainId)] = destinationToken;
-        else delete whitelistedRoutes[_whitelistedRouteKey(originChainId, originToken, destinationChainId)];
+        bool depositsEnabled
+    ) public override onlyOwner {
```

Comment:
One question I have is whether we should offer a `deleteWhitelistRoute` function that deletes this entire mapping (or just deletes the `destinationToken`) so that we can block pool rebalances. I view this as potentially less useful since only "invalid" pool rebalances can always be blocked at the UMIP level

Reply:
Wouldn't deleting be the same as calling this method with destinationToken set to 0x0 and depositsEnabled set to false?

Reply:
good point!